### PR TITLE
add solution

### PR DESCRIPTION
**Pull request by AndrewHudzenko**

### CODE REVIEW

File: src/main/java/mate/academy/Main.java [hunk 1]
```diff
@@ -1,16 +1,45 @@
...
+import mate.academy.model.MovieSession;
+import mate.academy.service.CinemaHallService;
 import mate.academy.service.MovieService;
+import mate.academy.service.MovieSessionService;
 
 public class Main {
-    public static void main(String[] args) {
-        MovieService movieService = null;
+    private static final Injector injector =
+            Injector.getInstance("mate.academy");
```
Comment (max-bolotin) [inline src/main/java/mate/academy/Main.java:15]: you might move package name to constant

File: src/main/java/mate/academy/model/MovieSession.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+@Table(name = "movie_sessions")
+public class MovieSession {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @ManyToOne
+    private Movie movie;
+    @ManyToOne
+    private CinemaHall cinemaHall;
+    private LocalDateTime showTime;
```
Comment (max-bolotin) [inline src/main/java/mate/academy/model/MovieSession.java:21]: it's better to name your columns correctly, for example name = "cinema_hall_id" and name = "show_time"

File: src/main/resources/hibernate.cfg.xml [hunk 3]
```diff
@@ -0,0 +1,15 @@
+<hibernate-configuration>
+    <session-factory>
+        <property name="hibernate.dialect">org.hibernate.dialect.MySQL5Dialect</property>
+        <property name="connection.url">jdbc:mysql://localhost:3306/ticket_app?serverTimeZone=UTC</property>
```
Comment (max-bolotin) [inline src/main/resources/hibernate.cfg.xml:4]: `?serverTimeZone=UTC` might not work, check if it's better to use `?serverTimezone=UTC`

**Review by max-bolotin — APPROVED** [review]

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> availableSessions = session.createQuery("from MovieSession ms"
+                    + " where ms.id = :id and ms.showTime between :day_start "
+                    + "and :day_finish", MovieSession.class);
+            availableSessions.setParameter("id", movieId);
+            availableSessions.setParameter("day_start", LocalDateTime.MIN);
+            availableSessions.setParameter("day_finish", LocalDateTime.MAX);
```
Comment (max-bolotin) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:57]: I'm not sure this will work, please check again

**Review by max-bolotin — APPROVED** [review]

File: src/main/java/mate/academy/dao/CinemaHallDao.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package mate.academy.dao;
+
+import java.util.List;
+import mate.academy.model.CinemaHall;
+
+public interface CinemaHallDao {
+    CinemaHall add(CinemaHall cinemaHall);
+
+    CinemaHall get(Long id);
```
Comment (Nick97-git) [inline src/main/java/mate/academy/dao/CinemaHallDao.java:9]: this method should return optional

File: src/main/java/mate/academy/dao/MovieSessionDao.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.dao;
+
+import java.time.LocalDate;
+import java.util.List;
+import mate.academy.model.MovieSession;
+
+public interface MovieSessionDao {
+    MovieSession add(MovieSession movieSession);
+
+    MovieSession get(Long id);
```
Comment (Nick97-git) [inline src/main/java/mate/academy/dao/MovieSessionDao.java:10]: this method should return optional

File: src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class CinemaHallDaoImpl implements CinemaHallDao {
+
```
Comment (Nick97-git) [inline src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java:13]: remove empty line

File: src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+import java.util.List;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+public class CinemaHallDaoImpl implements CinemaHallDao {
```
Comment (Nick97-git) [inline src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java:12]: where's Dao annotation?

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+import mate.academy.lib.Dao;
+import mate.academy.model.MovieSession;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieSessionDaoImpl implements MovieSessionDao {
+
```
Comment (Nick97-git) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:17]: remove empty line

File: src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<MovieSession> availableSessions = session.createQuery("from MovieSession ms"
+                    + " where ms.id = :id and ms.showTime between :day_start "
+                    + "and :day_finish", MovieSession.class);
+            availableSessions.setParameter("id", movieId);
+            availableSessions.setParameter("day_start", LocalDateTime.MIN);
+            availableSessions.setParameter("day_finish", LocalDateTime.MAX);
+            return availableSessions.getResultList();
+        } catch (Exception e) {
+            throw new DataProcessingException("Can't find available sessions with id "
```
Comment (Nick97-git) [inline src/main/java/mate/academy/dao/impl/MovieSessionDaoImpl.java:60]:
```suggestion
            throw new DataProcessingException("Can't find available sessions with movie's id "
```

File: src/main/java/mate/academy/model/MovieSession.java (hunk 2 above)
Comment (Nick97-git) [reply src/main/java/mate/academy/model/MovieSession.java:21]: agree with comment, let's do this

File: src/main/resources/hibernate.cfg.xml [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+        <property name="connection.username">root</property>
+        <property name="connection.password">admin</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="mate.academy.model.Movie"/>
+        <mapping class="mate.academy.model.CinemaHall"/>
+        <mapping class="mate.academy.model.MovieSession"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (Nick97-git) [inline src/main/resources/hibernate.cfg.xml:15]: add an empty line in the end of the file

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — APPROVED** [review]

**Pull request closed without merging**